src/main/docs/guide/appendix/architecture/aopArch.adoc

```diff
@@ -0,0 +1,123 @@
+<4> An internal array called `$interceptors` holds references to which api:aop.Interceptor[] instances apply to each method since an ann:aop.InterceptorBinding[] can be type or method level these may differ for each method.
+
+Each non-final and non-private method of the proxied type that has an ann:aop.InterceptorBinding[] associated with it (either type level or method level) is overridden with logic that proxies the original method, for example:
+
+[source,java]
+----
+@Overrides
+public void doWork(String taskName) {
+    ExecutableMethod method = this.$proxyMethods[0];
+    Interceptor[] interceptors = this.$interceptors[0]; // <1>
+    MethodInterceptorChain chain = new MethodInterceptorChain( // <2>
+        interceptors, 
+        this, 
+        method, 
+        new Object[]{taskName}
+    );
+    chain.proceed(); // <3>
+}
+----
+
+<1> The api:inject.ExecutableMethod[] and array of api:aop.Interceptor[] instances for the method is located.
+<2> A new api:aop.chain.MethodInterceptorChain[] is constructed with the interceptors, a reference to the intercepted instance, the method and the arguments.
+<3> The `proceed()` method is invoked on the api:aop.chain.MethodInterceptorChain[].
+
+Note that the default behaviour of the the ann:aop.Around[] annotation is to invoke methods on the intercepted super type that is subclassed so there is effectively only a single instance of the `NotNullExample` in the example above.
+
+However, this behaviour can be customized, for example using `@Around(proxyTarget=true)` will result two api:inject.BeanDefinition[] instances, one of the proxy and one for the proxy target. In this case the proxy becomes the primary bean and lookups up the proxy target and there are effectively two instances of the type.
+
+The default behaviour (`proxyTarget=false`) is more efficient memory wise as only a single api:inject.BeanDefinition[] is required.
+
+=== Security Considerations
+
+Method interception via `AROUND` advice is typically used to define logic that addresses cross cutting concerns, one of which is security.
+
+When multiple api:aop.Interceptor[] instances apply to a single method it may be important from a security perspective that these interceptors execute in a specific order.
+
+The api:aop.Interceptor[] interface extends the api:core.order.Ordered[] interface to enable the developer to control interceptor ordering by overriding the `getOrder()` method. 
+
+When the api:aop.chain.MethodInterceptorChain[] is constructed and multiple interceptors are present they are ordered with `HIGHEST` priority interceptors executed first.
+
+To aid the developer who defines their own <<aroundAdvice, Around Advice>> the api:aop.InterceptPhase[] enumeration defines various constants that can be used to correctly declare the value of `getOrder()` (for example security typically falls within the `VALIDATE` phase).
```

Comment:
Is there tooling/logging properties which would help dumping the interceptor chain to diagnose such ordering issues?